libethereum/Executive.cpp

```diff
@@ -180,6 +180,11 @@ void Executive::initialize(Transaction const& _transaction)
 {
 	m_t = _transaction;
 
+	BlockHeader bh;
+	bh.setNumber(m_envInfo.number());
+	bh.setGasLimit(m_envInfo.gasLimit());
+	m_sealEngine.verifyTransaction(ImportRequirements::Everything, m_t, bh);
```

Comment:
`Executive::initialize` saves the exceptions to `m_excepted` member for some reason, so I guess we need to do the same for exceptions thrown here from `verifyTransactions`

So we need here
1. Wrap this in try/catch, catching `Exception`, filling `m_excepted` and rethrowing
2. If you add catch clause for `OutOfGasIntrinsic`, you could remove the check in lines 197-204, because it's already checked in `verifyTransaction`. The catch clause should do the same as current check inside `if (m_baseGasRequired > m_t.gas())`